include/allegro5/inline/fmaths.inl

```diff
@@ -131,8 +131,7 @@ AL_INLINE(al_fixed, al_fixsub, (al_fixed x, al_fixed y),
          return 0x80000000;
       }
       else {
-         int res = lres >> 16;
-         return res;
+         return (al_fixed) (lres >> 16);
```

Comment:
Should this be instead `(al_fixed)lres >> 16`?  I know you only did this to fix a warning but the difference is your version does the bitshift in the `int` domain and then casts the result, versus converting `lres` to whatever type al_fixed is and *then* doing the shift.  I'm not sure it matters here but something worth taking into consideration.

Reply:
`al_fixed` is a typedef for `int32_t`. Here, `lres` is an `int64_t`. To multiply two `al_fixed` you need to multiply them 'as if' they were integers and then divide by 65536, equivalently shift right by 16.
So if you multiply, then cast to 32 bit, then right shift 16, the top 16 bits of your new `al_fixed` will be all 0 or all 1, i.e useless. You need to shift the full 64 bits, *then* cast to 32 bits.

Reply:
Maybe we could add some tests for fixed numbers (though they are not used much nowadays as I understand it)